src/operator/tensor/ordering_op-inl.h

```diff
@@ -284,29 +285,29 @@ MSHADOW_XINLINE void MergeTopK(index_t K, DType *val1, index_t *ind1, DType *val
   }
 }
 
-template<typename DType>
-__global__ void PartialSortSmallK(index_t K, index_t N, DType *val, index_t *ind, bool is_ascend) {
+template<typename DType, typename IDXType = int32_t>
+__global__ void PartialSortSmallK(IDXType K, IDXType N, DType *val, IDXType *ind, bool is_ascend) {
   // Buffer for pairwise reduction.
-  extern __shared__ index_t buff[];
+  extern __shared__ int32_t buff[];
   // Start of buffer sections associated with this thread.
-  const index_t offset(threadIdx.x*K);
-  index_t *ind_buff = &buff[offset];
+  const IDXType offset(threadIdx.x*K);
+  int32_t *ind_buff = &buff[offset];
```

Comment:
Since buff had to harfcoded to be int32 then this had to be done else error will be raised when compiler keeps IDXType=int64_t which raises an attempt to cast int32_t* to int64_t* (w/o explicit reinterpret_cast)